src/main/java/org/springframework/samples/petclinic/rest/VisitRestController.java

```diff
@@ -121,4 +124,16 @@
 		return new ResponseEntity<Void>(HttpStatus.NO_CONTENT);
 	}
 
+    @PreAuthorize( "hasRole(@roles.VET_ADMIN)" )
+    @RequestMapping(value = "/{visitId}/payment", method = RequestMethod.PUT, produces = "application/json")
+    public ResponseEntity<Visit> markPaid(@PathVariable("visitId") int visitId){
```

Comment:
Preferably, if we use PUT method, we should provide the visitor as part of the request body.
All the business logic must be inside the service layer.

We should either return the updated enity as it is defined in the method signature, or we must update it and return 204 - *No Content